Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -47,11 +67,26 @@ public struct SourceLocation: Hashable, Codable, CustomDebugStringConvertible {
   ///             location in the source file has `offset` 0.
   ///   - file: A string describing the name of the file in which this location
   ///           is contained.
-  public init(line: Int, column: Int, offset: Int, file: String) {
+  ///   - presumedLine: If the location has been adjusted using `#sourceLocation`
+  ///                   directives, the adjusted line. If `nil`, this defaults to
+  ///                   `line`.
+  ///   - presumedFile: If the location has been adjusted using `#sourceLocation`
+  ///                   directives, the adjusted file. If `nil`, this defaults to
+  ///                   `file`.
+  public init(
+    line: Int,
+    column: Int,
+    offset: Int,
+    file: String,
+    presumedLine: Int? = nil,
+    presumedFile: String? = nil
```

Comment:
Add to debug description?

Reply:
I am not super happy with the debug description as-is anyway because it also doesn’t include the file name and have been tempted to just remove it. WDYT?

Reply:
Or just extend it to include file as well 🤷

Reply:
Decided to remove `debugDescription`

> The debug description was lossy because it didn’t print file and offset. If we include that information, it’s so verbose that the debug description doesn’t really provide any benefit over the default debug description. Let’s just remove it.